.github/workflows/test-indexer-debian.yml

```diff
@@ -0,0 +1,42 @@
+name: Test Wazuh indexer on upgrade - Debian
+on:
+  push:
+    branches:
+      - 1850-wazuh-indexer-preserve-config-files-upon-upgrade-master
+  pull_request:
+    paths:
+      - 'stack/indexer/deb/debian/postrm'
+      - 'stack/indexer/deb/debian/postinst'
+      - 'stack/indexer/deb/debian/preinst'
+  workflow_dispatch:
+
+jobs:
+  Test-debian-indexer-upgrade:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download wazuh-indexer release package
+        run: |
+          curl -s https://packages.wazuh.com/key/GPG-KEY-WAZUH | gpg --no-default-keyring --keyring gnupg-ring:/usr/share/keyrings/wazuh.gpg --import && chmod 644 /usr/share/keyrings/wazuh.gpg
+          echo "deb [signed-by=/usr/share/keyrings/wazuh.gpg] https://packages.wazuh.com/4.x/apt/ stable main" | sudo tee -a /etc/apt/sources.list.d/wazuh.list
+          sudo apt-get update
+      - name: Get the release version of Wazuh
+        run: echo "RELEASE_VERSION=$(jq -r 'map(select(.prerelease == false and .draft == false)) | .[] | .tag_name' <<< $(curl --silent https://api.github.com/repos/wazuh/wazuh/releases) | sed "s|v||g" | sort -rV | head -n 1)" >> $GITHUB_ENV
+      - name: Get the major and minor of the release version
+        run: echo "MAJOR_MINOR=$(echo $RELEASE_VERSION | cut -d '.' -f1-2 | sed "s|\.||")" >> $GITHUB_ENV
+      - uses: actions/checkout@v3
+        with:
+          ref: 1850-wazuh-indexer-preserve-config-files-upon-upgrade-master
+      - name: Build wazuh-indexer issue-branch package and set env variable
+        working-directory: ./stack/indexer/deb
+        run: |
+          sudo ./build_package.sh
+          echo "PACKAGE_NAME=$(ls ./output | grep .deb | head -n 1)" >> $GITHUB_ENV
```

Comment:
To improve the reading of the file, consider adding an empty line between each task (name)

Reply:
Done.